### PR TITLE
Chore: Update ast to dst

### PR DESCRIPTION
**Pull request by spinillos**
It updates ast with dst but it depends of: https://github.com/grafana/thema/pull/98

### CODE REVIEW

File: public/locales/pseudo-LOCALE/grafana.json [hunk 1]
```diff
@@ -574,4 +574,4 @@
       "option-tooltip": "Cľęäř şęľęčŧįőŉş"
     }
   }
-}
```
Comment (spinillos) [inline public/locales/pseudo-LOCALE/grafana.json:577]: 🤔 idk why this is here

Comment (sdboyer) [reply public/locales/pseudo-LOCALE/grafana.json:577]: @joshhunt i see your name in the blame 😉 halp!

File: pkg/codegen/util_go.go [hunk 2]
```diff
@@ -13,32 +12,32 @@ import (
...
 	}
 
 	if cfg.walker != nil {
-		astutil.Apply(gf, cfg.walker, nil)
+		dstutil.Apply(gf, cfg.walker, nil)
 
 		err = format.Node(buf, fset, gf)
 		if err != nil {
-			return nil, fmt.Errorf("error formatting Go AST: %w", err)
+			return nil, fmt.Errorf("error formatting Go DST: %w", err)
```
Comment (sdboyer) [inline pkg/codegen/util_go.go:40]:
```suggestion
			return nil, fmt.Errorf("error formatting Go AST: %w", err)
```

File: pkg/codegen/jenny_gotypes.go [hunk 3]
```diff
@@ -7,13 +7,12 @@ import (
...
 	"github.com/grafana/codejen"
 	"github.com/grafana/thema/encoding/gocode"
-	"golang.org/x/tools/go/ast/astutil"
 )
 
 // GoTypesJenny creates a [OneToOne] that produces Go types for the provided
 // [thema.Schema].
 type GoTypesJenny struct {
-	ApplyFuncs []astutil.ApplyFunc
+	ApplyFuncs []dstutil.ApplyFunc
```
Comment (sdboyer) [inline pkg/codegen/jenny_gotypes.go:15]: With this change made, the body of the generator can be mostly eliminated b/c we can just pass `DecoderCompactor` as an `ApplyFuncs` item

File: pkg/plugins/plugindef/plugindef_types_gen.go [hunk 4]
```diff
@@ -67,6 +67,7 @@ const (
 )
 
 // Defines values for Category.
+// Defines values for PlugindefCategory.
```
Comment (sdboyer) [inline pkg/plugins/plugindef/plugindef_types_gen.go:70]: yeah, yuck, something going there with the switch to dst or the adapted version of the logic, clearly 😢

File: public/locales/pseudo-LOCALE/grafana.json (hunk 1 above)
Comment (joshhunt) [reply public/locales/pseudo-LOCALE/grafana.json:577]: Was yarn i18n:extract ran manually? (I can’t see a reason why it should have been). When was this branched from main (we merged that touched this file today?

You’re safe to just not commit this in your PR, and I can double check later that our lint/tests doesn’t make this file dirty in main at the moment.

**Review by sdboyer — APPROVED** [review]
Lovely!

Please post a follow-up issue to track down the problem with comment codegen we're seeing in plugindef so we don't forget. Once that's done, i think this is good to go 👍🏻

**Comment by spinillos** [comment]
Looks like that Enterprise error got cached because this bug was resolved here: https://github.com/grafana/grafana-enterprise/pull/4318

**Pull request merged**